tests/blackbox_test.go

```diff
@@ -0,0 +1,1215 @@
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			outer(t, test)
+		})
+	}
+}
+
+func PreCleanup(t *testing.T) {
+	mountpoints, _ := exec.Command(
+		"findmnt", "-l", "-o", "target").CombinedOutput()
+	points := strings.Split(string(mountpoints), "\n")
+	for i := len(points) - 1; i >= 0; i-- {
+		pat := "/tmp/hd1p*"
+		match, err := filepath.Match(pat, points[i])
+		if err != nil {
+			t.Log(err)
+		}
+		if match {
+			_, _ = exec.Command("umount", points[i]).CombinedOutput()
+		}
+	}
+	removeFile(t, "config.ign")
+	removeFile(t, "test.img")
+}
+
+func outer(t *testing.T, test Test) {
+	PreCleanup(t)
+	t.Log(test.name)
+
+	path := os.Getenv("PATH")
+	_ = os.Setenv("PATH", fmt.Sprintf("%s:%s", "bin/amd64/tests", path))
+
+	imgName := "test.img"
+	imageSize := calculateImageSize(test.in)
+
+	// Finish data setup
+	for _, part := range test.in {
+		if part.GUID == "" {
+			part.GUID = generateUUID(t)
+		}
+		updateTypeGUID(t, part)
+	}
+	setOffsets(test.in)
+	for _, part := range test.out {
+		updateTypeGUID(t, part)
+	}
+	setOffsets(test.out)
+
+	// Creation
+	createVolume(t, imgName, imageSize, 20, 16, 63, test.in)
+	setDevices(t, imgName, test.in)
+	mountRootPartition(t, test.in)
+	if strings.Contains(test.config, "passwd") {
+		prepareRootPartitionForPasswd(t, test.in)
+	}
+	mountPartitions(t, test.in)
+	createFiles(t, test.in)
+	unmountPartitions(t, test.in)
+
+	// Ignition
+	config := test.config
+	for _, d := range test.mntDevices {
+		device := pickDevice(t, test.in, imgName, d.label)
+		config = strings.Replace(config, d.code, device, -1)
+	}
+	writeIgnitionConfig(t, config)
+	root := getRootLocation(t, test.in)
+	runIgnition(t, "disks", root)
+	runIgnition(t, "files", root)
+
+	// Update out structure with mount points & devices
+	setExpectedPartitionsDrive(test.in, test.out)
+
+	// Validation
+	mountPartitions(t, test.out)
+	validatePartitions(t, test.out, imgName)
+	validateFiles(t, test.out)
+
+	// Cleanup
+	_ = os.Setenv("PATH", path)
+	unmountPartitions(t, test.out)
+	unmountRootPartition(t, test.out)
+	removeMountFolders(t, test.out)
+	removeFile(t, "config.ign")
+	removeFile(t, imgName)
+}
+
+func prepareRootPartitionForPasswd(t *testing.T, partitions []*Partition) {
+	for _, p := range partitions {
+		if p.Label == "ROOT" {
+			_ = os.MkdirAll(filepath.Join(p.MountPath, "home"), 0755)
+			_ = os.MkdirAll(filepath.Join(p.MountPath, "usr", "bin"), 0755)
+			_ = os.MkdirAll(filepath.Join(p.MountPath, "usr", "sbin"), 0755)
+			_ = os.MkdirAll(filepath.Join(p.MountPath, "usr", "lib64"), 0755)
+			_ = os.MkdirAll(filepath.Join(p.MountPath, "etc"), 0755)
+
+			_ = os.Symlink(
+				filepath.Join(p.MountPath, "usr", "lib64"), filepath.Join(
+					p.MountPath, "lib64"))
+			_ = os.Symlink(
+				filepath.Join(p.MountPath, "usr", "bin"), filepath.Join(
+					p.MountPath, "bin"))
+			_ = os.Symlink(
+				filepath.Join(p.MountPath, "usr", "sbin"), filepath.Join(
+					p.MountPath, "sbin"))
+
+			_, _ = exec.Command(
```

Comment:
All of the libraries being pulled in are for Ignition itself. If they aren't present the `getpwnam_r` in `internal/exec/util/user_group_lookup.c` on line 61 will set `pptr` to null.